### ISSUE TITLE
Use GitHub's fork of `cmark`

### ISSUE THREAD

**Issue by RoyiAvital** [mike-ward/Markdown-Edit#213]
Hello,

I saw that GutHub released their for of [`mark`](https://github.com/github/cmark).
It seems to be the fastest engine out there.

Some writing about it:
http://ropensci.org/blog/blog/2016/12/02/commonmark

Any chance using it on MarkDownEdit?

Thank You.

**Comment by nonplayer** [comment]
Another vote for this. I'm forced to disable github in the settings because of some _really_ weird rendering bug that still doesn't make any sense to me. However, that means I can't use tables instead as the price paid for turning off github rendering.

Switching to this fork of cmark would fix that issue.

**Comment by mike-ward** [comment]
Pandoc does differ from GitHub in several ways. Sure would be nice if they published C# bindings for their fork. I'll add it to the to-do list.

**Comment by mike-ward** [comment]
I'm guessing it would. Kind of annoyed they don't have a binary release. Still needs C# bindings. Thinking I might do that.

**Comment by veremenko-jav** [comment]
According to python bindings you only need to import cmark_markdown_to_html function

**Comment by mike-ward** [comment]
Oh if that were only true. The extensions (like tables, strikethrough) have to be loaded separately. See the cmark.cs file and look at main.c

My current thinking is to just run the cmark.exe program with the correct arguments and stream the result. Much like what is done with Pandoc elsewhere.

**Comment by veremenko-jav** [comment]
Yes, I tried to add it as well as adding it to "CustomMarkdownConverter", tables didn't work. Building is not difficult though. You just need cmake.
